spec/Section 3 -- Type System.md

```diff
@@ -2085,3 +2091,91 @@ to the relevant IETF specification.
 ```graphql example
 scalar UUID @specifiedBy(url: "https://tools.ietf.org/html/rfc4122")
 ```
+
+### @defer
+
+```graphql
+directive @defer(
+  label: String
+  if: Boolean
+) on FRAGMENT_SPREAD | INLINE_FRAGMENT
+```
+
+The `@defer` directive may be provided for fragment spreads and inline fragments
+to inform the executor to delay the execution of the current fragment to
+indicate deprioritization of the current fragment. A query with `@defer`
+directive will cause the request to potentially return multiple responses, where
+non-deferred data is delivered in the initial response and data deferred is
+delivered in a subsequent response. `@include` and `@skip` take precedence over
+`@defer`.
+
+```graphql example
+query myQuery($shouldDefer: Boolean) {
+   user {
+     name
+     ...someFragment @defer(label: 'someLabel', if: $shouldDefer)
+   }
+}
+fragment someFragment on User {
+  id
+  profile_picture {
+    uri
+  }
+}
+```
+
+#### @defer Arguments
+
+- `if: Boolean` - When `true`, fragment _should_ be deferred. When `false`,
+  fragment will not be deferred and data will be included in the initial
+  response. If omitted, defaults to `true`.
+- `label: String` - A unique label across all `@defer` and `@stream` directives
```

Comment:
Should there be a formal validation rule for the the label unicity? For instance it could clarify whether labels should be unique in an operation + any fragments it references:

```graphql
query Query {
   ... F1 @defer(label: "x")  # <- unique in this operation
}

fragment F1 on Query {
   ... F2 @defer(label: "x")  # <- but not really unique
}
```
I think the example above should be an error but not 100% sure?

Reply:
@BoD i moved this to it's own discussion: https://github.com/robrichard/defer-stream-wg/discussions/35